README.md

```diff
@@ -447,6 +448,181 @@ you can create your own Configuration(s) and compose them with Config's ++ opera
 
 Then you can build as usual with CONFIG=MyConfig.
 
+## <a name="debug"></a> Debugging with GDB
+
+### 1) Generating the RBB emulator
+
+The objective of this section is to use GNU debugger to debug programs RISC-V programs running on the emulator in the same fashion as in [Spike](https://github.com/riscv/riscv-isa-sim#debugging-with-gdb).
+
+For that we need to build a Remote Bit Bang enabled emulator, this could be any of the listed configurations in `rocket-chip/src/main/scala/system/Configs.scala`. Basically all we need is to extend our configuration with JtagDTMSystem that will deal with program loading and interface with the debugger. In the following example we added this extention to the DefaultConfig one.
+
+    class DefaultConfigRBB extends Config(
+    new WithJtagDTMSystem ++ new WithNBigCores(1) ++ new BaseConfig)
+
+    class QuadCoreConfigRBB extends Config(
+    new WithJtagDTMSystem ++ new WithNBigCores(4) ++ new BaseConfig)
+
+To build the configuration we use the command:
+
+    rocket-chip$ cd emulator
+    emulator$ CONFIG=DefaultConfigRBB make
+
+By default the emulator is generated under the name `emulator-freechips.rocketchip.system-DefaultConfigRBB`.
+
+### 2) Compiling the executable
+
+We suppose that `helloworld` is our program, you can use `crt.S`, `syscalls.c` and the linker script `test.ld` to construct your own program, check examples stated in [riscv-tests](https://github.com/riscv/riscv-tests).
+
+In our case we will use the following example:
+
+```
+char text[] = "Vafgehpgvba frgf jnag gb or serr!";
+
+// Don't use the stack, because sp isn't set up.
+volatile int wait = 1;
+
+int main()
+{
+    while (wait)
+        ;
+
+    // Doesn't actually go on the stack, because there are lots of GPRs.
+    int i = 0;
+    while (text[i]) {
+        char lower = text[i] | 32;
+        if (lower >= 'a' && lower <= 'm')
+            text[i] += 13;
+        else if (lower > 'm' && lower <= 'z')
+            text[i] -= 13;
+        i++;
+    }
+
+    while (!wait)
+        ;
+}
+```
+Do not forget to compile with `-g -Og` flags to provide debugging support just as explained [here](https://github.com/riscv/riscv-isa-sim#debugging-with-gdb).
+
+### 3) Launch the emulator
+
+We can then launch the emulator with
+
+    ./emulator-freechips.rocketchip.system-DefaultConfigRBB +jtag_rbb_enable=1 --rbb-port=9823 helloworld
+	This emulator compiled with JTAG Remote Bitbang client. To enable, use +jtag_rbb_enable=1.
+	Listening on port 9823
+	Attempting to accept client socket
+
+### 4) Launch OpenOCD
+
+We first need to ensure the OpenOCD has been generated. Generally it is in `$(RISCV)/bin/openocd`. We then define a configuration file that is going to define the RBB port we will use, which is in our case `9823`.
```

Comment:
Edited version:

You will need RISC-V Enabled OpenOCD binary. This is installed with `riscv-tools` in `$(RISCV)/bin/openocd`, or can be compiled manually from [riscv-openocd](https://github.com/riscv/riscv-openocd). OpenOCD requires a configuration file, in which we define the RBB port we will use, which is in our case `9823`.

Reply:
Done